### PR TITLE
Fix #25392: Template Builder Add Row With System Container When Template Is Empty

### LINKED ISSUE #25392: Template Builder: Add Default Single Row to New Templates
### Problem Statement

Currently, when trying to add a new template in dotCMS, no default single row is added. This leads to an empty template without any row configured, causing inconvenience to users.

You need to add a new row in order to add a new box. Adding a row by default can reduce the number of clicks to create a new template. 

image

### Steps to Reproduce

- Log in to dotCMS.
- Go to the Templates section.
- Click on "Add Template".
- Provide a name for the new template.
- Try to add a new box

### Acceptance Criteria

- After adding a new template, a default single row should be automatically added to the template.
- The single row should be visible when opening the newly created template.

### dotCMS Version

Tested on master // Docker // FF 

### Proposed Objective

Quality Assurance

### Proposed Priority

Priority 2 - Important

### PR DESCRIPTION
**Pull request by zJaaal**
### Summary
:sparkles::white_check_mark::recycle:

Improved the handling of empty layouts in the template builder component. Added a constant `EMPTY_ROWS_VALUE` to represent the default gridstack layout and used it in the `gridstack-utils.ts` and `gridstack-utils.spec.ts` files. Added new test cases to cover the empty layout scenarios.

> _No rows, no mercy, only `EMPTY_ROWS_VALUE`_
> _We mock the grid with a subgrid of doom_
> _We test the edge cases, we parse the dot object_
> _We import the constant, we improve the condition_

### Walkthrough
*  Import and define `EMPTY_ROWS_VALUE` constant to represent default gridstack layout ([link](https://github.com/dotCMS/core/pull/25397/files?diff=unified&w=0#diff-3647ec392c96563c5712834ab3a47c47021cc2e0b7aa88a8add5bd49849a6a42L4-R4), [link](https://github.com/dotCMS/core/pull/25397/files?diff=unified&w=0#diff-f3792a46b295fb618b6d4a8d8f4ecef72ec5e3c41a470d6a44886881a6b93659R5-R6), [link](https://github.com/dotCMS/core/pull/25397/files?diff=unified&w=0#diff-98de89ac988dbfd17c912bb493096714d7c139835179a99162b4df2eb3c309e5L6-R39))
*  Modify `parseFromDotObjectToGridStack` function to return `EMPTY_ROWS_VALUE` when body parameter is undefined or has empty rows array ([link](https://github.com/dotCMS/core/pull/25397/files?diff=unified&w=0#diff-f3792a46b295fb618b6d4a8d8f4ecef72ec5e3c41a470d6a44886881a6b93659L116-R118))
*  Add test cases for `parseFromDotObjectToGridStack` function to check edge cases with undefined or empty body parameter ([link](https://github.com/dotCMS/core/pull/25397/files?diff=unified&w=0#diff-3647ec392c96563c5712834ab3a47c47021cc2e0b7aa88a8add5bd49849a6a42L43-R53))

## Screenshots 
https://github.com/dotCMS/core/assets/63567962/715a33c9-25ac-44ac-9e11-7f7cc1dc79a4

### CODE REVIEW

File: core-web/libs/template-builder/src/lib/components/template-builder/utils/gridstack-utils.ts [hunk 1]
```diff
@@ -113,9 +115,7 @@ export function removeColumnByID(row: DotGridStackWidget, columnID: string): Dot
 export function parseFromDotObjectToGridStack(
     body: DotLayoutBody | undefined
 ): DotGridStackWidget[] {
-    if (!body) {
-        return [];
-    }
+    if (!body?.rows.length || !body) return EMPTY_ROWS_VALUE;
```
Comment (fmontes) [inline core-web/libs/template-builder/src/lib/components/template-builder/utils/gridstack-utils.ts:118]:
```suggestion
    if (!body || !body?.rows.length) return EMPTY_ROWS_VALUE;
```

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by danielmdob — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

**Review by fmontes — APPROVED** [review]
